docs/administering/OFFBOARDING_A_FUND.md

```diff
@@ -0,0 +1,170 @@
+=> [ApplicationRecord(abstract)]
+
+# Catch all the model counts for each fund to eyeball after
+starting_counts = {}
+Fund.all.sort.each do |fund|
+  fund_counts = {}
+  models_fund_dependent.each do |model|
+    fund_counts[model.to_s] = model.where(fund_id: fund.id).size
+  end
+  starting_counts[fund.name] = fund_counts
+end
+fund_counts = {}
+models_fund_agnostic.each do |model|
+  fund_counts[model.to_s] = model.all.size
+end
+starting_counts['fund_agnostic'] = fund_counts
+
+#save starting counts to file
+$stdout = File.new('initail_state.out', 'w')
+$stdout.sync = true
+starting_counts
+$stdout = STDOUT
+```
+
+Remove tenant-specific rows.
+```ruby
+#Lock our tenant!
+Fund.all.pluck(:name,:id)
+defunct_fund_id = #the fund you are removing
+ActsAsTenant.current_tenant = Fund.find(defunct_fund_id)
+
+Patient.destroy_all
+ArchivedPatient.destroy_all
+[Note, Fulfillment, ExternalPledge, PracticalSupport, Call].each { |model| model.destroy_all }
+[Clinic, Line, Config].each { |model| model.destroy_all }
+User.destroy_all
+
+# exit and restart to console to drop the ActsAsTenant.current_tenant
+exit
+```
+
+
+Check everything looks right before destroying the PaperTrail versions
+
+Run `rails c`
+```ruby
+# Check the new overall state
+models_fund_dependent = [ PledgeConfig, PracticalSupport, Patient, Note, Line, Fulfillment, ExternalPledge, Event, Config, Clinic, CallListEntry, Call, ArchivedPatient, User]
+models_fund_agnostic = [ Fund, ActiveRecord::SessionStore::Session, PaperTrail::Version, PaperTrailVersion] 
+
+post_deletion_counts = {}             
+Fund.all.sort.each do |fund|          
+  fund_counts = {}                    
+  models_fund_dependent.each do |model|
+    fund_counts[model.to_s.to_sym] = model.where(fund_id: fund.id).size
+  end                                                                  
+  post_deletion_counts[fund.name] = fund_counts                        
+end                                                                    
+fund_counts = {}                                                       
+models_fund_agnostic.each do |model|                                   
+  fund_counts[model.to_s.to_sym] = model.all.size                      
+end                                                                    
+post_deletion_counts['fund_agnostic'] = fund_counts                    
+                                                                       
+#save starting counts to file                                          
+$stdout = File.new('post_deletion.out', 'w')                           
+$stdout.sync = true                                                    
+post_deletion_counts                                                   
+$stdout = STDOUT
+
+exit
+```
+
+diff those files!
+
+  - The other fund objects should not have changed significantly
+  - There should be a jump in PaperTrails corresponding to the total object count we've deleted
+
+`diff -y initial_count.out post_deletion.out`
```

Comment:
I think these commands aren't plopping the right stuff in the files, though maybe I have something configured weird?

Reply:
what are you getting? I see a structure like:

```ruby
{
  "fund a" => {
              "Notes" => 400
              "Patient" => 500
               ...
              }
   "fund b" => { 
             "Notes" => 20
              "Patient" => 40
               ...
               }
     "fund_agnostic" => {
          "Funds" => 4
          "Sessions" => 500
          "PaperTrail" => 40904
}
```